Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActExcelTests.cs

```diff
@@ -0,0 +1,213 @@
+﻿using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.Common.InterfacesLib;
+using Amdocs.Ginger.CoreNET.ActionsLib;
+using Amdocs.Ginger.CoreNET.Repository;
+using GingerCore.Actions;
+using GingerCoreNETUnitTest.RunTestslib;
+using GingerTestHelper;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Data;
+using System.IO;
+using System.Linq;
+
+namespace GingerCoreNETUnitTest.LinuxTransformationTests
+{
+    [TestClass]
+    public class ActExcelTests
+    {
+        string excelPathRead = TestResources.GetTestResourcesFile(@"Excel" + Path.DirectorySeparatorChar + "Names.xlsx");
+        string excelPathWrite = TestResources.GetTestResourcesFile(@"Excel" + Path.DirectorySeparatorChar + "NamesWrite.xlsx");
+        string excelPathWriteTemp = TestResources.GetTempFile("NamesWriteTemp.xlsx");
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            WorkSpace.Init(new WorkSpaceEventHandler());
+            WorkSpace.Instance.SolutionRepository = GingerSolutionRepository.CreateGingerSolutionRepository();
+            File.Copy(excelPathWrite, excelPathWriteTemp,true);
+        }
+        
+        [TestMethod]
+        public void ReadExcelFirstRowTest()
+        {
+            //Arrange            
+            ActExcel actExcel = new ActExcel();
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue(nameof(ActExcel.ExcelFileName),
+                TestResources.GetTestResourcesFile(@"Excel" + Path.DirectorySeparatorChar + "Names.xlsx"));
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue(nameof(ActExcel.SheetName), "Sheet1");
+            actExcel.ExcelActionType = ActExcel.eExcelActionType.ReadData;
+            actExcel.AddNewReturnParams = true;
+
+            //Act
+            actExcel.Execute();
+
+            //Assert
+            Assert.AreEqual(actExcel.ActReturnValues.Count,4);
+        }
+        [TestMethod]
+        public void ReadExcelFirstRowWithFilterTest()
+        {
+            //Arrange            
+            ActExcel actExcel = new ActExcel();
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue(nameof(ActExcel.ExcelFileName),
+                TestResources.GetTestResourcesFile(@"Excel" + Path.DirectorySeparatorChar + "Names.xlsx"));
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue(nameof(ActExcel.SheetName), "Sheet1");
+            actExcel.AddOrUpdateInputParamValueAndCalculatedValue(nameof(ActExcel.SelectRowsWhere), "Last='Cohen'");
+            actExcel.ExcelActionType = ActExcel.eExcelActionType.ReadData;
+            actExcel.AddNewReturnParams = true;
+
+            //Act
+            actExcel.Execute();
+
+            //Assert
+            Assert.AreEqual(string.Join(',',actExcel.ActReturnValues.Select(x=> x.Actual).ToList()), "1,Mark,Cohen,2109 Fox Dr");
```

Comment:
this is not enough to validate only the "Actual" field, need to validate also the "Param" and "Path"

Reply:
add param